lib/ansible/modules/packaging/language/pip.py

```diff
@@ -235,6 +241,72 @@
                              'pip': 'import pkg_resources; print(pkg_resources.get_distribution("pip").version)'}
 
 
+class Distribution(object):
+    def __init__(self, name_string, version_string=None):
+        if not name_string.startswith(('svn+', 'git+', 'hg+', 'bzr+', 'file:')):
+            self._plain_distribution = True
+            if version_string:
+                version_string = version_string.lstrip()
+                separator = '==' if version_string[0].isdigit() else ' '
+                name_string = separator.join((name_string, version_string))
+            self._requirement = Requirement.parse(name_string)
+            self._distribution_name = self._requirement.name.lower()
+        else:
+            self._plain_distribution = False
+            self._distribution_name = name_string
+
+    @property
+    def has_version_specifier(self):
+        if self._plain_distribution:
+            return bool(self._requirement.specs)
+        return False
+
+    @property
+    def distribution_name(self):
+        return self._distribution_name
+
+    @distribution_name.setter
+    def distribution_name(self, new_name):
+        self._distribution_name = new_name
+
+    def is_satisfied_by(self, version_to_test):
+        if self._plain_distribution:
+            return self._requirement.specifier.contains(version_to_test)
+        return True
```

Comment:
This is tricky. It should probably return false because there's no way of knowing whether it's the same version, so we'll need to invoke pip to reinstall it as it does when you run it manually